phoenix-core/src/main/java/org/apache/phoenix/util/ScanUtil.java

```diff
@@ -363,10 +363,12 @@ public static void setTimeRange(Scan scan, long minStamp, long maxStamp) {
         }
         int[] position = new int[slots.size()];
         int maxLength = 0;
+        int slotEndingFieldPos = 0;
         for (int i = 0; i < position.length; i++) {
             position[i] = bound == Bound.LOWER ? 0 : slots.get(i).size()-1;
             KeyRange range = slots.get(i).get(position[i]);
-            Field field = schema.getField(i + slotSpan[i]);
+            slotEndingFieldPos = slotEndingFieldPos + slotSpan[i] + (i>0? 1: 0);
```

Comment:
nit:

This is quite confusing : `+ (i>0? 1: 0);` .

Can we increment slotEndingFieldPos at the end of the for cycle body instead?
Or just initialize slotEndingFieldPos to -1 , and increment it here unconditionally ?

Reply:
👍🏼